exporter/datadogexporter/traces_exporter.go

```diff
@@ -64,6 +67,7 @@ func newTracesExporter(
 		sourceProvider:   sourceProvider,
 		retrier:          clientutil.NewRetrier(params.Logger, cfg.BackOffConfig, scrubber),
 		metadataReporter: metadataReporter,
+		metricsClient:    datadog.NewMetricClient(params.MeterProvider.Meter("datadogexporter")),
```

Comment:
Do we really need this as a field in the `traceExporter`? We just need to initialize it, but we don't need to store it here.